nav/filter_lidar_data/src/ignore_lidar_node.cpp

```diff
@@ -0,0 +1,113 @@
+#include <boost/bind/placeholders.hpp>
+#include <tf/transform_listener.h>
+#include <geometry_msgs/TransformStamped.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geographic_msgs/GeoPoint.h>
+#include <geodesy/utm.h>
+#include <limits>
+
+#include "../include/ignore_lidar_node.h"
+
+IgnoreLidarNode::IgnoreLidarNode(ros::NodeHandle nh)
+    : nh_{nh}
+{
+    ROS_INFO("Connected to Node");
+
+    // Transform Gps waypoints into pose
+    geometry_msgs::PoseStamped second_pose_stamped = getPoseFromGps(second_waypoint_.latitude, 
+                                                            second_waypoint_.longitude, "/odom");
+    ROS_INFO("second_waypoint transformed: x: %f, y: %f", second_pose_stamped.pose.position.x,
+                                                          second_pose_stamped.pose.position.y);
+    geometry_msgs::PoseStamped third_pose_stamped = getPoseFromGps(third_waypoint_.latitude, 
+                                                            third_waypoint_.longitude, "/odom");
+    ROS_INFO("third_waypoint transformed: x: %f, y: %f", third_pose_stamped.pose.position.x,
+                                                          third_pose_stamped.pose.position.y);
+    
+    // Set rectangular bounds where lidar data needs to be adjusted. Based on the 2nd and 3rd waypoint 
+    bounds_[0].x = second_pose_stamped.pose.position.x + 1.5;
+    bounds_[0].y = second_pose_stamped.pose.position.y + 4.0;
+    bounds_[1].x = third_pose_stamped.pose.position.x - 1.0;
+    bounds_[1].y = third_pose_stamped.pose.position.y - 4.0;
+
+    // Subscribe to recieve lidar data and current pose
+    lidar_sub_ = nh_.subscribe("/scan", 10, &IgnoreLidarNode::lidarCallback, this);
+    odom_sub_ = nh_.subscribe("/odometry/global", 10, &IgnoreLidarNode::odomCallback, this);
+
+    // Publish to a new topic containing filtered lidar data
+    ignore_lidar_pub_ = nh_.advertise<sensor_msgs::LaserScan>("/scan_modified", 1);
+
+}    
+
+void IgnoreLidarNode::lidarCallback(const sensor_msgs::LaserScanConstPtr& lidar_msg)
+{
+    if (!is_odom_init) {
+        return; // Do not do calculations without a pose to work with
+    }
+
+    if (cur_odom_.x <= bounds_[0].x && 
+        cur_odom_.x >= bounds_[1].x && 
+        cur_odom_.y <= bounds_[0].y && 
+        cur_odom_.y >= bounds_[1].y) {
+
+        ROS_INFO("At Second Waypoint");
+        sensor_msgs::LaserScan output_msg = *lidar_msg; // Make a copy of lidar_msg
+
+        // Set each element in ranges to inf (No obstacles present) to clear costmap
+        double inf = std::numeric_limits<double>::infinity();
```

Comment:
Is this a bug? I think the message expects a float but we'll take it since it works I guess